src/multiPortConnector.js

```diff
@@ -0,0 +1,241 @@
+/*
+ * Copyright 2022, Tony Atkins
+ * Copyright 2011-2020, Colin Clark
+ *
+ *  Licensed under the MIT license, see LICENSE for details.
+ */
+
+(function (fluid) {
+    "use strict";
+    var youme = fluid.registerNamespace("youme");
+
+    // Enclosing grade so that we can recreate our pool of connections.
+    fluid.defaults("youme.multiPortConnector.connections", {
+        gradeNames: ["fluid.component"],
+
+        invokers: {
+            open: {
+                funcName: "youme.multiPortConnector.callAllChildInvokers",
+                args: ["{that}", "youme.connection", "open"] // that, gradeName, invokerName, invokerArgs
+            },
+
+            close: {
+                funcName: "youme.multiPortConnector.callAllChildInvokers",
+                args: ["{that}", "youme.connection", "close"] // that, gradeName, invokerName, invokerArgs
+            }
+        }
+    });
+
+    fluid.defaults("youme.multiPortConnector.connections.inputs", {
+        gradeNames: ["youme.multiPortConnector.connections"],
+
+        dynamicComponents: {
+            connection : {
+                type: "youme.connection.input",
+                sources: "{connections}.options.sources",
+                options: {
+                    openImmediately: true,
+                    portSpec: "{source}.portSpec",
+                    members: {
+                        port: "{source}.port"
+                    }
+                }
+            }
+        }
+    });
+
+    fluid.defaults("youme.multiPortConnector.connections.outputs", {
+        gradeNames: ["youme.multiPortConnector.connections"],
+
+        dynamicComponents: {
+            connection : {
+                type: "youme.connection.output",
+                sources: "{connections}.options.sources",
+                options: {
+                    openImmediately: true,
+                    portSpec: "{source}.portSpec",
+                    members: {
+                        port: "{source}.port"
+                    }
+                }
+            }
+        }
+    });
+
+    fluid.defaults("youme.multiPortConnector", {
+        gradeNames: ["fluid.modelComponent"],
+
+        events: {
+            attemptConnection: null
+        },
+
+        model: {
+            portSpecs: [],
+            ports: "{youme.system}.model.ports"
+        },
```

Comment:
This does indeed look like a classic case where lensed components would save a lot of risk and heartache. What I suggest you do is recast the "attemptConnection" function as a model relay into a fresh area of the model named "connectionSources" and then lens the "connection" dynamicComponents from "{that}.model.connectionSources". Should then let you axe the component visitor, and the extra level of components, and the "createOnEvent" annotations.

Reply:
You might want to try simply destroying the enclosing component and rely on the natural destroy semantics to destroy and hence close all the children automatically.

Reply:
I believe this has been addressed by the lensing stuff and modelised open/close stuff.